releasenotes-builder/src/main/java/com/github/checkstyle/Main.java

```diff
@@ -94,7 +95,8 @@ public static void main(String... args) {
         }
         else {
             System.out.printf("%nGeneration ends with %d errors.%n",
-                              errorCounter);
+                              errors.size());
+            printListOf(errors);
```

Comment:
old logic is very complicated that confuse us a lot.

we need simply:
if exception is thrown, put content of `ex.printStackTrace();` to `errors`. `hasThrownException` will not be required.
if case `error` is not empty we print all errors and exit with error code.

if `publicationErrors` exists they are also erros, and it should fail execution also. Such old logic on `publicationErrors` might exist due to old problem to do publication and this is how we tried to suppress them, as we couldnot resolve them. Now we resolved all problems and publications must work and must fail execution if any problems happens.

Reply:
Facing some trouble adding this stack trace to the errors. Apparently [there is no easy way to convert stack trace to string](https://stackoverflow.com/questions/1149703/how-can-i-convert-a-stack-trace-to-a-string)

We either do this
```
StringWriter sw = new StringWriter();
PrintWriter pw = new PrintWriter(sw);
e.printStackTrace(pw);
String sStackTrace = sw.toString(); // stack trace as a string
```

or use [org.apache.commons.lang.exception.ExceptionUtils.getStackTrace(Throwable)](http://commons.apache.org/proper/commons-lang/javadocs/api-2.6/org/apache/commons/lang/exception/ExceptionUtils.html#getStackTrace(java.lang.Throwable)).

Both of them do not pass checkstyle so we have to allow one. Which do you prefer?
```
[ERROR] [checkstyle] [ERROR] /media/stoyan/Elements/Open Source/contribution/releasenotes-builder/src/main/java/com/github/checkstyle/Main.java:27:1: Disallowed import - org.apache.commons.lang.exception.ExceptionUtils. [ImportControlMain]
```
```
[ERROR] [checkstyle] [ERROR] /media/stoyan/Elements/Open Source/contribution/releasenotes-builder/src/main/java/com/github/checkstyle/Main.java:23:1: Disallowed import - java.io.PrintWriter. [ImportControlMain]
[ERROR] [checkstyle] [ERROR] /media/stoyan/Elements/Open Source/contribution/releasenotes-builder/src/main/java/com/github/checkstyle/Main.java:24:1: Disallowed import - java.io.StringWriter. [ImportControlMain]

```

Reply:
Can we print exception as is and put in error messages string like "exception happened, se details above "

Reply:
We can do that, yes.

Reply:
Done.